lib/src/facts/external/windows/powershell_resolver.cc

```diff
@@ -49,19 +56,44 @@ namespace facter { namespace facts { namespace external {
                 }
             }
 
-            string error;
-            each_line(
-                pwrshell,
-                {
-                    "-NoProfile",
-                    "-NonInteractive",
-                    "-NoLogo",
-                    "-ExecutionPolicy",
-                    "Bypass",
-                    "-File",
-                    file
-                },
-                [&facts](string const& line) {
+            auto const ps_opts = vector<string> {
+                "-NoProfile",
+                "-NonInteractive",
+                "-NoLogo",
+                "-ExecutionPolicy",
+                "Bypass",
+                "-File",
+                file
+            };
+
+            auto result = execute(pwrshell, ps_opts, 0, {
+                execution_options::trim_output,
+                execution_options::merge_environment,
+                execution_options::throw_on_failure
+            });
+
+            try {
+                auto node = YAML::Load(result.output);
```

Comment:
We're relying on the YAML implementation to parse JSON?

Reply:
Right - this is pretty much a copy-and-paste from the execution resolver, which does the same thing for other types of executables

Reply:
For the case of static facts when we know the file type, it makes sense to use a fast JSON parser when we can. When we're already testing multiple formats and doing fallback, adding a third parser into the mix seemed like the wrong tradeoff. That was my thinking when I did the initial YAML/JSON for executable facts, at any rate

Reply:
That seems fine.